scripts/generate-releases.sh

```diff
@@ -2,64 +2,84 @@
-snapshot_list="$(kubectl -n rh-acs-tenant get snapshot.appstudio.redhat.com -l pac.test.appstudio.openshift.io/sha="${COMMIT}" -o json | jq -r '
-  .items
-  | map(select((.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="'"${BRANCH}"'") or (.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="refs/heads/'${BRANCH}'")))
-  | sort_by(.spec.application)
-  | group_by(.spec.application)
-  | map(sort_by(.metadata.creationTimestamp) | last)
-  | .[]
-  | "\(.metadata.name)|\(.spec.application)"
-')"
-
-validate_input() {
+get_snapshots() {
+    commit="$1"
+    branch="$2"
+    kubectl get -n rh-acs-tenant snapshot -l pac.test.appstudio.openshift.io/sha="${commit}" -o json | jq -r '
+        .items
+        | map(select((.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="'"${branch}"'") or (.metadata.annotations["pac.test.appstudio.openshift.io/source-branch"]=="refs/heads/'${branch}'")))
+        | sort_by(.spec.application)
+        | group_by(.spec.application)
+        | map(sort_by(.metadata.creationTimestamp) | last)
+        | .[]
+        | "\(.metadata.name)|\(.spec.application)"'
+}
+
+# Validate all expected Snapshots were found.
+validate_snapshots() {
+    commit="$1"
+    snapshot_list="$2"
     pipelines_count="$(find ".tekton" -maxdepth 1 -type f -name "operator-index-ocp-*-build.yaml" | wc -l )"
     snapshots_count="$(echo "$snapshot_list" | wc -l )"
-    echo -e "found the following snapshots for \033[0;32m$COMMIT\033[0m commit in \033[0;32m$BRANCH\033[0m branch:" >&2
+
+    echo -e "found the following snapshots for \033[0;32m$commit\033[0m commit:" >&2
     echo "$snapshot_list" >&2
 
     if [[ "$snapshots_count" -eq 0 ]]; then
-        echo "ERROR: Could not find any Snapshots for the commit '${COMMIT}'." >&2
-        exit 1
-    fi
-    if [[ "${ENVIRONMENT}" != "staging" && "${ENVIRONMENT}" != "prod" ]]; then
-        echo "ERROR: ENVIRONMENT input must either be 'staging' or 'prod'." >&2
-        exit 1
-    fi
-    if [[ "${ENVIRONMENT}" == "prod" && "${BRANCH}" != "master" ]]; then
-        echo "ERROR: prod release has to be done on master branch" >&2
-        exit 1
+        echo "ERROR: Could not find any Snapshots for the commit '${commit}'." >&2
+        return 1
     fi
     if [[ "$snapshots_count" -ne "$pipelines_count" ]]; then
-        echo "ERROR: The number of snapshots ($snapshots_count) for $COMMIT in branch $BRANCH does not match the number of supported OCP versions ($pipelines_count)." >&2
-        exit 1
+        echo "ERROR: The number of snapshots $snapshots_count for $commit does not match the number of supported OCP versions $pipelines_count." >&2
+        return 1
     fi
 }
 
+# Generate the Release resources for each snapshot found.
 generate_release_resources() {
+    environment="$1"
+    snapshot_list="$2"
+    release_name="${environment}-$(git rev-parse --short HEAD)-$(date +'%Y-%m-%d-%H-%M')"
```

Comment:
Converting to `local`-s, plus declarations for variables used in the loop
```suggestion
    local -r environment="$1"
    local -r snapshot_list="$2"

    local release_name
    release_name="${environment}-$(git rev-parse --short HEAD)-$(date +'%Y-%m-%d-%H-%M')"

    local snapshot
    local application
    local release_plan
```